xo/geometry/catmull_rom.h

```diff
@@ -70,16 +70,21 @@ namespace xo
 		{
 			xo_assert( points.size() >= 4 && times.size() == ( points.size() - 2 ) );
 			segments_.reserve( points.size() - 4 );
-			for ( size_t i = 0; i <= points.size() - 4 )
+			for ( size_t i = 0; i <= points.size() - 4; ++i)
 				segments_.emplace_back( points[ i ], points[ i + 1 ], points[ i + 2 ], points[ i + 3 ] );
 		}
 
 		P operator()( T t ) {
+			//assume times_ is sorted
 			auto it = std::lower_bound( times_.begin(), times_.end(), t );
-			if ( it == times_.end() ) it = times_.begin();
-			if ( it == times_.end() - 1 ) it = times_.end() - 2;
-			size_t index = it - times_.begin();
-			return segments_[ index ]( ( t - *it ) / ( *( it + 1 ) - *it ) );
```

Comment:
The index computation here is incorrect. 
The lower_bound return the iterator pointing to the first element which does not compare less than the input t.
Considering times_ as [0, 0.5, 1]:
1) if t = 0.1, the lower bound is 0.5 (instead of 0), index is 1 instead of 0;
2) if t = 1.2, it==times_.end(), the code has the index as 0 instead 1;